backend/src/main/java/com/huellapositiva/domain/model/entities/ContactPerson.java

```diff
@@ -20,6 +27,13 @@ public ContactPerson(Id accountId, EmailAddress emailAddress, PasswordHash passw
         super(accountId, emailAddress, passwordHash, id);
     }
 
+    public ContactPerson(Id accountId, EmailAddress emailAddress, PasswordHash passwordHash, Id id, RegisterESALMemberRequestDto dto) {
+        super(accountId, emailAddress, passwordHash, id);
```

Comment:
I didn't know that a User had two different Ids, but I've seen that the first is the id of the credential, and the second is the id of the account itself. So, in both User, Contact Person and Volunteer, their Ids are poorly named.

Reply:
A mi no me lo parece que estén mal nombrados, tiene sentido. Cuando tu creas un usuario siempre tiene 2 ID, uno que relacionado con la información en credenciales y otro que está relacionado con la creación de su tipo específico de usuario. Ten en cuenta que cada registro en cada tabla tiene su ID específico, hasta location. Al momento de crear un usuario tienes que crear 2 tablas (mínimo). Luego lo que veo es que se está haciendo uso del poder de la herencia algo que me parece adecuado... quizás no tanto por el hecho de tener información de las propiedades en el padre como "name", "surname" o el "id" del contact person, pero el "id" de la credencial si me parece acertado.

¿A lo mejor necesita un refactor? Posiblemente, pero en esta rama no, almenos sobre lo que he hecho si puedo refactorizar algo.

¿Qué opinas @ibaiul?

Reply:
Cada vez que se crea un User, recibe un "accountId" y un "id" a secas. Cuando tenemos en cuenta que uno es de la credencial y el otro del usuario en sí, no queda claro cuál de los dos es "accountId". De hecho, hasta que no lo revisé y vi que accountId es de la credencial, yo suponía lo contrario. Creo que si se llamaran credentialId y accountId o userId habría menor posibilidad de confusión.

Reply:
Pues se cambia el nombre, mañana lo cambio, ahora es muy tarde.

Reply:
Hecho